File: backend/app/schemas/report.py
"""
backend/app/schemas/report.py
Pydantic schemas for Executive BI Report Generator.
"""
from typing import List, Dict, Any, Optional
from pydantic import BaseModel


class KPIScorecard(BaseModel):
    title: str
    value: str
    status: str                             # "POSITIVE", "NEUTRAL", "CONCERN"
    trend_note: str


class ExecutiveInsight(BaseModel):
    category: str                           # "Growth", "Risk", "Efficiency", "Data Quality"
    finding: str
    impact: str
    recommendation: str


class ExecutiveReportRequest(BaseModel):
    dataset_id: str
    custom_instructions: Optional[str] = None


class ExecutiveReportResponse(BaseModel):
    dataset_id: str
    dataset_name: str
    generated_at: str
    executive_summary: str
    kpi_scorecards: List[KPIScorecard]
    insights: List[ExecutiveInsight]
    strategic_recommendations: List[str]
    markdown_report: str

File: frontend/types/report.ts
// frontend/types/report.ts

export interface KPIScorecard {
  title: string;
  value: string;
  status: "POSITIVE" | "NEUTRAL" | "CONCERN";
  trend_note: string;
}

export interface ExecutiveInsight {
  category: "Growth" | "Risk" | "Efficiency" | "Data Quality";
  finding: string;
  impact: string;
  recommendation: string;
}

export interface ExecutiveReportRequest {
  dataset_id: string;
  custom_instructions?: string;
}

export interface ExecutiveReportResponse {
  dataset_id: string;
  dataset_name: string;
  generated_at: string;
  executive_summary: string;
  kpi_scorecards: KPIScorecard[];
  insights: ExecutiveInsight[];
  strategic_recommendations: string[];
  markdown_report: string;
}
